erpnext/selling/doctype/sales_order/sales_order.py

```diff
@@ -831,7 +831,7 @@ def update_item(source, target, source_parent):
 
 	for supplier in suppliers:
 		po = frappe.get_list("Purchase Order", filters={"sales_order":source_name, "supplier":supplier, "docstatus": ("<", "2")})
-		if len(po) == 0:
+		if len(po) == 0 or any( item.get("delivered_by_supplier") == 1 for item in selected_items):
```

Comment:
selected_items is fetched from the table in the dialog (client side) right ? From what i know `delivered_by_supplier` is not sent in the selected items. So `item.get("delivered_by_supplier")` will always be `None`.

Anyway we know that `selected_items` here contains drop ship items only, this method is specifically for that.
Partial Drop ship wasnt considered before. 
Lets remove line 833 and 834 (if condition) altogether.
Remove `else: suppliers =[]` as well and bring all the logic outside the if else. Lets just map whatever we get to the Purchase Order.
Logic to set `qty` as pending qty is there in  `update_item`. So our mapping will be correct in case of partial.